servo/servo.py

```diff
@@ -435,6 +479,50 @@ def top_level_schema_json(self, *, all: bool = False) -> str:
             default=pydantic.json.pydantic_encoder,
         )
 
+    async def connect(self) -> None:
+        # must define dynamic function to allow configurable backoff
+        await backoff.on_exception(
+            backoff.expo,
+            httpx.HTTPError,
+            max_time=lambda: self.config.settings.backoff.max_time(),
+            max_tries=lambda: self.config.settings.backoff.max_tries(),
+            on_giveup=lambda details: self.logger.critical(
```

Comment:
max_tries=lambda: self.config.settings.backoff.max_tries()

should be equivalent to:
 max_tries=self.config.settings.backoff.max_tries
 
 (there doesn't seem to be any need to define a function that just calls another function with the same exact signature - might as well just pass the reference to the latter)

Reply:
The lambda encloses a reference to the original servo so if the config is altered dynamically the updated config will be applied instead of the decorator using a stale method of a possibly garbage collected config object (that said this level of dynamicness is not currently in use)